Rename SpanContext to SpanReference. (#1075)
open-telemetry/opentelemetry-specification: specification/overview.md

- A set of zero or more **Events**, each of which is itself a tuple (timestamp, name, [**Attributes**](./common/common.md#attributes)). The name must be strings.
- Parent's **Span** identifier.
- [**Links**](#links-between-spans) to zero or more causally-related **Spans**
  (via the **SpanReference** of those related **Spans**).

> - **SpanReference** identification of a Span. See below.

Reviewer: ```suggestion
- **SpanReference** information required to reference a Span. See below.
```

### SpanReference

> Represents all the information that identifies **Span** in the **Trace** and

Reviewer: Just a suggestion:

```suggestion
Represents all the information that is required to reference a **Span** in the **Trace** unambiguously and efficiently and
```

E.g. in Dynatrace we could reference the span with just span+trace ID but require some routing info in the trace state to do so efficiently. Not sure if this is generally applicable enough.

Author: Definitely something that can be done in a another PR, specially because this was not *even* touched in this PR.

Reviewer: I think this one is more related to that PR, as the name change reflects a change in the understanding of that type. But I'm fine with doing this in a follow-up too (I already approved)

MUST be propagated to child Spans and across process boundaries. A

> **SpanReference** contains the tracing identifiers and the options that are

Reviewer: ```suggestion
**SpanReference** contains the tracing identifiers, tracestate and the options that are
```

Author: Same, the lack of trace state is *not* related to this PR.

### Links between spans

A **Span** may be linked to zero or more other **Spans** (defined by
**SpanReference**) that are causally related. **Links** can point to

> **SpanReferences** inside a single **Trace** or across different **Traces**.

Reviewer: ```suggestion
**Spans** inside a single **Trace** or across different **Traces**.
```

Links contain SpanReferences but semantically, they *point to* actual Spans.

## Propagators

OpenTelemetry uses `Propagators` to serialize and deserialize cross-cutting concern values

> such as `SpanReference` and `Baggage`. Different `Propagator` types define the restrictions

Reviewer: ```suggestion
such as `Span`s (usually only the `SpanReference` portion) and `Baggage`. Different `Propagator` types define the restrictions
```
